### PR TITLE
Update diagnostics for duplicated product/target names

### PR DESCRIPTION
**Pull request by neonichu**
rdar://127623219

### CODE REVIEW

File: Tests/WorkspaceTests/WorkspaceTests.swift [hunk 1]
```diff
@@ -13015,17 +13015,9 @@ final class WorkspaceTests: XCTestCase {
         do {
             workspace.sourceControlToRegistryDependencyTransformation = .disabled
 
-            XCTAssertThrowsError(try workspace.checkPackageGraph(roots: ["root"]) { _, diagnostics in
-                testDiagnostics(diagnostics) { result in
```
Comment (neonichu) [inline Tests/WorkspaceTests/WorkspaceTests.swift:13019]: This was unreachable.

File: Tests/WorkspaceTests/WorkspaceTests.swift [hunk 2]
```diff
@@ -13135,31 +13127,16 @@ final class WorkspaceTests: XCTestCase {
 
         do {
             workspace.sourceControlToRegistryDependencyTransformation = .disabled
-            if ToolsVersion.current < .v5_8 {
```
Comment (neonichu) [inline Tests/WorkspaceTests/WorkspaceTests.swift:13138]: This was obviously always false.

**Comment by neonichu** [comment]
@swift-ci please test

**Review by xedin — APPROVED** [review]
Thank you!

**Comment by xedin** [comment]
@neonichu could you please cherry-pick this to 6.0 branch as well?

**Comment by neonichu** [comment]
Sure https://github.com/apple/swift-package-manager/pull/7575

**Pull request merged**